testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/settings/SettingsMenuActivity.kt

```diff
@@ -0,0 +1,139 @@
+            enableSaveView()
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        settings = AppSettings.instance
+        setContentView(R.layout.settings_menu_activity)
+
+        initializeActionBar()
+        settingsProgressDialog = SettingsProgressDialog(this)
+        renderAppSettingsScreen()
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        menuInflater.inflate(R.menu.settings_menu, menu)
+        menu.findItem(R.id.settings_menu_save).isEnabled = saveViewEnabled
+        return true
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            android.R.id.home -> {
+                finish()
+                return true
+            }
+            R.id.settings_menu_save -> {
+                onSaveAction()
+                return true
+            }
+            else -> super.onOptionsItemSelected(item)
+        }
+    }
+
+    private fun onSaveAction() {
+        settingsProgressDialog.show()
+
+        updateSettings(
+            editAppId.text.toString(),
+            editSubscriptionKey.text.toString(),
+            switchTestMode.isChecked
+        )
+    }
+
+    private fun initializeActionBar() {
+        val toolBar = findViewById<Toolbar>(R.id.toolbar)
+        setSupportActionBar(toolBar)
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+        supportActionBar?.setDisplayShowHomeEnabled(true)
+    }
+
+    private fun renderAppSettingsScreen() {
+        textInfo.text = createSettingsInfo()
+        editAppId.setText(settings.appId)
+        editSubscriptionKey.setText(settings.subscriptionKey)
+        switchTestMode.isChecked = settings.isTestMode
+
+        editAppId.addTextChangedListener(settingsTextWatcher)
+        editSubscriptionKey.addTextChangedListener(settingsTextWatcher)
+
+        enableSaveView()
+    }
+
+    private fun createSettingsInfo(): String {
+        return "Build " + getString(R.string.miniapp_sdk_version) + " - " +
+                getString(R.string.build_version)
+    }
+
+    private fun enableSaveView() {
+        saveViewEnabled =
+            !(editAppId.text.toString().isInvalidUuid() || editSubscriptionKey.text.isNullOrEmpty())
+    }
+
+    private fun updateSettings(appId: String, subscriptionKey: String, isTestMode: Boolean) {
+        val appIdHolder = settings.appId
+        val subscriptionKeyHolder = settings.subscriptionKey
+        val isTestModeHolder = settings.isTestMode
+        settings.appId = appId
+        settings.subscriptionKey = subscriptionKey
+        settings.isTestMode = isTestMode
+
+        launch {
+            try {
+                MiniApp.instance(AppSettings.instance.miniAppSettings).listMiniApp()
+                settings.isSettingSaved = true
+                runOnUiThread {
+                    settingsProgressDialog.cancel()
+                    finish()
```

Comment:
There is a reason why we use `recreate` before.
Please take a look at this as research. It would help you to get more understanding of the source code.